protocols.py

```diff
@@ -1,32 +1,45 @@
 from binaryninja import (BinaryView, BackgroundTask, HighLevelILCall, RegisterValueType, HighLevelILAddressOf,
                          HighLevelILVar, Constant, Function, HighLevelILVarSsa, HighLevelILVarInitSsa,
-                         TypeFieldReference, bundled_plugin_path, log_info, log_warn, log_alert)
-from typing import Optional, Tuple
+                         TypeFieldReference, bundled_plugin_path, log_info, log_warn, log_alert, user_plugin_path)
+from typing import Optional, Tuple, List
 import os
 import sys
 import struct
 
 protocols = None
 
-def init_protocol_mapping():
-    # Parse EFI definitions only once
-    global protocols
-    if protocols is not None:
-        return True
 
-    # Find the EFI type definition file within the Binary Ninja installation
+def init_protocol_mapping(bv: BinaryView):
     if sys.platform == "darwin":
-        efi_def_path = os.path.join(bundled_plugin_path(), "..", "..", "Resources", "types", "efi.c")
+        bundled_efi_path = os.path.join(bundled_plugin_path(), "..", "..", "Resources", "types", "efi.c")
+        user_efi_path = os.path.join(user_plugin_path(), "..", "..", "Resources", "types", "platform", f"{bv.platform.name}.c")
     else:
-        efi_def_path = os.path.join(bundled_plugin_path(), "..", "types", "efi.c")
+        bundled_efi_path = os.path.join(bundled_plugin_path(), "..", "types", "efi.c")
+        user_efi_path = os.path.join(user_plugin_path(), "..", "types", "platform", f"{bv.platform.name}.c")
```

Comment:
Debating on whether we should use `{bv.platform.name}.c` or just use `efi.c`. I think just using `efi.c` will be easier. I get there are sometimes architecture specific types, but for the most part I think the user is just going to use this mechanism for protocol structs that are unlikely to be architecture-specific.

Reply:
The problem here is the types should be parsed and introduced to binary view. The `efi.c` file inside the user folder won't be automatically parsed.